### PR TITLE
Ports timer cooldowns + bonus fix to abductor implants

### PR DESCRIPTION
**Pull request by Crossedfall**
## About The Pull Request 

Ports:

https://github.com/tgstation/tgstation/pull/51388
https://github.com/tgstation/tgstation/pull/51763

Bonus port: https://github.com/tgstation/tgstation/pull/46447 (required for the above to work)

closes #3751 

## Why It's Good For The Game

Enables future ports, it's also a pretty clean system all things considered

## Changelog
:cl:Rohesie, LemonInTheDark, spookydonut
code: Ports the TGMC timer cooldowns system.
fix: Fixed stealth implants not properly boxing the user, and doors not shocking
refactor: Renamed COOLDOWN_CHECK to COOLDOWN_FINISHED
fix: Abductor implant now reports the correct remaining time until it can be used again.
/:cl:

### CODE REVIEW

**Comment by Vexylius** [comment]
You've mentioned about it porting https://github.com/tgstation/tgstation/pull/51763 twice. Did you mean to mention this https://github.com/tgstation/tgstation/pull/51388?

**Comment by Crossedfall** [comment]
No issues during TM

File: code/controllers/subsystem/timer.dm [hunk 1]
```diff
@@ -516,6 +516,21 @@ SUBSYSTEM_DEF(timer)
...
+// How long left on a timer
+/proc/timeleft(id)
+	if (!id)
+		return null
+	if (id == TIMER_ID_NULL)
+		CRASH("Tried to get timeleft of a null timerid. Use TIMER_STOPPABLE flag")
+	if (!istext(id))
+		if (istype(id, /datum/timedevent))
+			var/datum/timedevent/timer = id
+			return timer.timeToRun - world.time
```
Comment (ike709) [inline code/controllers/subsystem/timer.dm]:
```suggestion
	if (istype(id, /datum/timedevent))
		var/datum/timedevent/timer = id
		return timer.timeToRun - world.time
```

File: code/game/objects/items/implants/implant_abductor.dm [hunk 2]
```diff
@@ -5,22 +5,17 @@
 	icon_state = "implant"
 	activated = 1
 	var/obj/machinery/abductor/pad/home
-	var/cooldown = 30
+	var/cooldown = 60 SECONDS
+	var/on_cooldown
```
Comment (ike709) [inline code/game/objects/items/implants/implant_abductor.dm]: I don't see this being set to true anywhere?

Comment (Crossedfall) [reply code/game/objects/items/implants/implant_abductor.dm]: This variable is set on line 18

**Review by francinum — DISMISSED** [review]

**Comment by Crossedfall** [comment]
Going to update the abductor code to use the new cooldowns just for consistency

File: code/game/objects/items/implants/implant_abductor.dm (hunk 2 above)
Comment (Crossedfall) [reply code/game/objects/items/implants/implant_abductor.dm]: Swapped to the new system, so this is no longer relevant

**Pull request merged**